### PR TITLE
Issue #1070 - Lambda Plugin Optional Role ARN Prop

### PR DESCRIPTION
**Pull request by Ethan-99**
Adds the capability to provide an optional `roleArn` property to the EntityAWSLambdaOverviewCard so it can assume said role when getting credentials to read the lambda(s).

The use case is that we have a service that has lambdas deployed in several different AWS accounts (one account per environment), and using this method, we can have multiple Lambda overview cards per service, each looking for the lambdas in different AWS accounts.

Also updates the README for the aws auth plugin to correct an incorrect example for requesting cross-account role access.

#### :heavy_check_mark: Checklist

- [ ] Added tests for new functionality and regression tests for bug fixes
- [ ] Added changeset (run `yarn changeset` in the root)
- [ ] Screenshots of before and after attached (for UI changes)
- [X ] Added or updated documentation (if applicable)

### CODE REVIEW

**Review by Xantier — APPROVED** [review]

**Comment by Xantier** [comment]
Could you update the branch and add patch notes to the PR

**Comment by Ethan-99** [comment]
> Could you update the branch and add changeset to the PR?

@Xantier Ran the `yarn changeset` workflow and merged in main. It appears these changes are not passing the merge check tests however and I am unsure why. Any chance you would be willing to look?

File: .changeset/clever-comics-carry.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@roadiehq/backstage-plugin-aws-lambda': patch
```
[the remark this answers is not in the document]
Comment (Ethan-99) [reply .changeset/clever-comics-carry.md]: updated, thanks

**Comment by Xantier** [comment]
Looks to be failing some tests

**Comment by Ethan-99** [comment]
> Looks to be failing some tests

Yeah, I'm not sure what's going on there.

The e2e test seems to be failing when getting AWS credentials, despite these changes not adjusting the backend credentials api: `CypressError: Timed out retrying after 30000ms: `cy.wait()` timed out waiting `30000ms` for the 1st request to the route: `getAwsCredentials`. No request ever occurred.`

And the verify test may also be related to an inability to auth in order to get the mock lambda, but I don't know.

Do you have any suggestions? Is there a way to run these tests locally?

**Review by Xantier — APPROVED** [review]
Thanks!

**Pull request merged**